### PR TITLE
wp-now: Mount the parent theme when running in a child theme

### PR DESCRIPTION
**Pull request by kozer**
## What?

This pr introduces a fix when the user works in a child theme. In this case, the parent theme is not mounted and this creates a "Missing parent theme" error:
child_theme_error

## Why?

This pr is the first attempt to fix that error.
There is plenty of discussion about this problem [here](https://github.com/WordPress/playground-tools/issues/33)

## How?

As a first step to solve this issue, we try to find the parent theme and mount it in the appropriate folder.
We assume that the parent theme is available in the parent directory, and there is some upcoming work to handle more complex scenarios

## Testing Instructions

1. Create a folder with a theme (eg: test), and a child theme (eg: test-child)
2. Run `wp-now start --path=<your-child-theme-path>`
3. Ensure there is no error in the loaded page
4. Move to `/wp-admin/themes.php` and ensure that you:
     - Can activate and deactivate your child theme
     - The parent theme is visible and can be activated and deactivated as well

### CODE REVIEW

File: packages/wp-now/src/wp-now.ts [hunk 1]
```diff
@@ -236,6 +237,16 @@ async function runPluginOrThemeMode(
 		projectPath,
 		`${documentRoot}/wp-content/${directoryName}/${pluginName}`
 	);
+	if (mode === WPNowMode.THEME) {
+		const templateName = getThemeTemplate(projectPath);
+		if (templateName) {
```
Comment (danielbachhuber) [inline packages/wp-now/src/wp-now.ts:242]: Should we make sure the directory exists before mounting?

File: packages/wp-now/src/wp-now.ts [hunk 2]
```diff
@@ -318,6 +329,15 @@ async function activatePluginOrTheme(
 	}
 }
 
+function getThemeTemplate(projectPath: string) {
```
Comment (danielbachhuber) [inline packages/wp-now/src/wp-now.ts]: Can you add some tests for this new function?

File: packages/wp-now/src/wp-now.ts [hunk 3]
```diff
@@ -236,6 +237,16 @@ async function runPluginOrThemeMode(
 		projectPath,
 		`${documentRoot}/wp-content/${directoryName}/${pluginName}`
 	);
+	if (mode === WPNowMode.THEME) {
```
Comment (danielbachhuber) [inline packages/wp-now/src/wp-now.ts:240]: Can you add some new tests for this behavior?

File: packages/wp-now/src/wp-now.ts (hunk 1 above)
Comment (adamziel) [reply packages/wp-now/src/wp-now.ts:242]: Mount() does that now

**Review by danielbachhuber — APPROVED** [review]
Works as expected with one tweak: a479df31b58babe7142adabcc7bb7a4333017f79

I downloaded https://wordpress.org/themes/moina-blog/, unzipped it, and ran `nx preview wp-now start --path=/Users/danielbachhuber/projects/playground-test/child-theme/moina-blog`. It was marked as a broken theme because it was missing its parent:

image

When I downloaded the parent theme (https://wordpress.org/themes/moina/) and unzipped it in the same directory, the child theme activated as expected:

image

**Pull request merged**